File: backend/src/routes/blog.ts
import { Router } from 'express';

const router = Router();

// Sample in-memory data for posts
const posts = [
  { id: 1, title: 'First Post', content: 'Hello World!' },
  { id: 2, title: 'Building a Terminal App', content: 'Using node-pty and xterm.js...' }
];

// GET /api/blog
router.get('/', (req, res) => {
  res.json({ success: true, data: posts });
});

// GET /api/blog/:id
router.get('/:id', (req, res) => {
  const post = posts.find(p => p.id === parseInt(req.params.id));
  if (post) {
    res.json({ success: true, data: post });
  } else {
    res.status(404).json({ success: false, error: 'Post not found' });
  }
});

export default router;

File: api/blog.go
package api

import (
	"net/http"
	"strconv"

	"github.com/gin-gonic/gin"
)

type BlogPost struct {
	ID      int    `json:"id"`
	Title   string `json:"title"`
	Content string `json:"content"`
}

var posts = []BlogPost{
	{ID: 1, Title: "First Post", Content: "Hello World!"},
	{ID: 2, Title: "Building a Terminal App", Content: "Using node-pty and xterm.js..."},
}

func RegisterBlogRoutes(router *gin.RouterGroup) {
	blog := router.Group("/blog")
	{
		blog.GET("/", func(c *gin.Context) {
			c.JSON(http.StatusOK, gin.H{"success": true, "data": posts})
		})

		blog.GET("/:id", func(c *gin.Context) {
			idStr := c.Param("id")
			id, err := strconv.Atoi(idStr)
			if err != nil {
				c.JSON(http.StatusBadRequest, gin.H{"success": false, "error": "Invalid ID"})
				return
			}

			for _, post := range posts {
				if post.ID == id {
					c.JSON(http.StatusOK, gin.H{"success": true, "data": post})
					return
				}
			}

			c.JSON(http.StatusNotFound, gin.H{"success": false, "error": "Post not found"})
		})
	}
}
